docs/unified-ingress.md

```diff
@@ -0,0 +1,107 @@
+# Unified Ingress Configuration
+
+This document describes the unified ingress approach implemented in the eoAPI Helm chart.
+
+## Overview
+
+eoAPI now uses a consolidated, controller-agnostic ingress configuration. This approach:
+
+- Eliminates code duplication between different ingress controller implementations
+- Provides consistent behavior across controllers
+- Simplifies testing and maintainability
+- Removes artificial restrictions on using certain ingress controllers in specific environments
+- Makes it easier to add support for additional ingress controllers in the future
+
+## Configuration
+
+The ingress configuration has been streamlined and generalized in the `values.yaml` file:
+
+```yaml
+ingress:
+  # Unified ingress configuration for both nginx and traefik
+  enabled: true
+  # ingressClassName: "nginx" or "traefik"
+  className: "nginx"
+  # Path configuration
+  pathType: "Prefix"  # Can be "Prefix" or "ImplementationSpecific" based on controller
+  pathSuffix: ""      # Add a suffix to service paths (e.g. "(/|$)(.*)" for nginx regex)
+  rootPath: ""        # Root path for doc server
+  # Host configuration
+  host: ""
+  # Custom annotations to add to the ingress
+  annotations: {}
+  # TLS configuration
+  tls:
+    enabled: false
+    secretName: eoapi-tls
+    certManager: false
+    certManagerIssuer: letsencrypt-prod
+    certManagerEmail: ""
+```
+
+## Controller-Specific Configurations
+
+### NGINX Ingress Controller
+
+For NGINX, use the following configuration:
+
+```yaml
+ingress:
+  enabled: true
+  className: "nginx"
+  pathType: "Prefix"
+  annotations:
+    nginx.ingress.kubernetes.io/use-regex: "true"
+    nginx.ingress.kubernetes.io/enable-cors: "true"
+    nginx.ingress.kubernetes.io/enable-access-log: "true"
+```
+
+### Traefik Ingress Controller
+
+When using Traefik, the system automatically includes the Traefik middleware to strip prefixes (e.g., `/stac`, `/raster`) from requests before forwarding them to services. This is handled by the `traefik-middleware.yaml` template.
+
+For basic Traefik configuration:
+
+```yaml
+ingress:
+  enabled: true
+  className: "traefik"
+  pathType: "Prefix"
+  # When using TLS, setting host is required to avoid "No domain found" warnings
+  host: "example.domain.com"  # Required to work properly with TLS
+  annotations:
+    traefik.ingress.kubernetes.io/router.entrypoints: web
+```
+
+For Traefik with TLS:
+
+```yaml
+ingress:
+  enabled: true
+  className: "traefik"
+  pathType: "Prefix"
+  # Host is required when using TLS with Traefik
+  host: "example.domain.com"
+  annotations:
+    traefik.ingress.kubernetes.io/router.entrypoints: websecure
+  tls:
+    enabled: true
+    secretName: eoapi-tls
+```
+
+## Migration
+
+If you're migrating from a previous version, follow these guidelines:
```

Comment:
Should we specify which versions need to follow the migration guide? something like "if you're migrating from a version <= 0.6.0 ..."

Reply:
I will specify it in this doc. My intent is that in the new [documentation](https://github.com/developmentseed/eoapi-k8s/issues/217), we have a comprehensive migration guide